### ISSUE TITLE
Can't use any other dbhost than localhost

### ISSUE THREAD

**Issue by RealSebFox** [fisharebest/webtrees#4543]
Hello. 

I have an issue with the folowing configuration: 

db.example.com hosts my database.
web.example.com hosts my webfiles.
webtrees.example.com is my virtual host.

This is the content of my config.ini.php:

```
; <?php return; ?> DO NOT DELETE THIS LINE
dbtype="mysql"
dbhost="db.example.com"
dbport="3306"
dbuser="username"
dbpass="secret"
dbname="databasename"
tblpfx="wt_"
base_url="https://webtrees.example.com"
rewrite_urls="0"
```

The database is present and can be viewed in PHPmyAdmin. When I call webtrees in my browser, I get the following error:

```
RuntimeException: SQLSTATE[HY000] [1045] Access denied for user 'username'@'web.example.com' (using password: YES) in /var/www/webtrees.example.com/app/Http/Middleware/UseDatabase.php:115 Stack trace: #0 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseDatabase->process() #1 /var/www/webtrees.example.com/app/Http/Middleware/BadBotBlocker.php(233): Middleland\Dispatcher->handle() #2 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\BadBotBlocker->process() #3 /var/www/webtrees.example.com/app/Http/Middleware/CompressResponse.php(73): Middleland\Dispatcher->handle() #4 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\CompressResponse->process() #5 /var/www/webtrees.example.com/app/Http/Middleware/ContentLength.php(40): Middleland\Dispatcher->handle() #6 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\ContentLength->process() #7 /var/www/webtrees.example.com/vendor/middlewares/client-ip/src/ClientIp.php(65): Middleland\Dispatcher->handle() #8 /var/www/webtrees.example.com/app/Http/Middleware/ClientIp.php(47): Middlewares\ClientIp->process() #9 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\ClientIp->process() #10 /var/www/webtrees.example.com/app/Http/Middleware/HandleExceptions.php(90): Middleland\Dispatcher->handle() #11 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\HandleExceptions->process() #12 /var/www/webtrees.example.com/app/Http/Middleware/BaseUrl.php(79): Middleland\Dispatcher->handle() #13 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\BaseUrl->process() #14 /var/www/webtrees.example.com/app/Http/Middleware/ReadConfigIni.php(68): Middleland\Dispatcher->handle() #15 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\ReadConfigIni->process() #16 /var/www/webtrees.example.com/app/Http/Middleware/SecurityHeaders.php(48): Middleland\Dispatcher->handle() #17 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\SecurityHeaders->process() #18 /var/www/webtrees.example.com/app/Http/Middleware/EmitResponse.php(57): Middleland\Dispatcher->handle() #19 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\EmitResponse->process() #20 /var/www/webtrees.example.com/vendor/oscarotero/middleland/src/Dispatcher.php(118): Middleland\Dispatcher->handle() #21 /var/www/webtrees.example.com/app/Webtrees.php(275): Middleland\Dispatcher->dispatch() #22 /var/www/webtrees.example.com/app/Webtrees.php(262): Fisharebest\Webtrees\Webtrees::dispatch() #23 /var/www/webtrees.example.com/index.php(51): Fisharebest\Webtrees\Webtrees->httpRequest() #24 {main}
```

It's very strange that the database host in the error message is the webhost instead of the host configured in my "dbhost" parameter. 

I've also tried using the local IP address of the database host and that gives me the same error. 

I've tried PHP 7.4, PHP 8.0 and PHP 8.1. I have tried MOD PHP, PHP CLI and PHP FPM.

When I copy the database to web.example.com and I use "localhost" or "127.0.0.1" as dbhost, it works. 

It seems webtrees is looking up the wrong host when I don't use localhost for my database. 

Thank you!

**Comment by fisharebest** [comment]
You understand that mysql user accounts include a host, so username@web.example is different to username@webtrees.example.

The error message comes from mysql, so we ARE connecting to the server.

Run `select user,host from mysql user` to see what accounts are set up.

**Comment by RealSebFox** [comment]
> You understand that mysql user accounts include a host, so [username@web.example](mailto:username@web.example) is different to [username@webtrees.example](mailto:username@webtrees.example).
> 
> The error message comes from mysql, so we ARE connecting to the server.
> 
> Run `select user,host from mysql user` to see what accounts are set up.

Hello,

Yes, I understand that. Nonetheless, the host in the error message is the wrong host. It's supposed to be 'username'@'db.example.com' instead of 'username'@'web.example.com'. That is something I still don't understand.

However, I did find a missing permission in the database configuration. I hadn't checked that before, because the error indicates we're connecting to the wrong server. That error put me on the wrong foot. When I fixed the missing permission, it worked. 

I'd still like to know why that error shows the wrong host, but the most important thing is, it works!

FYI: We just discovered Webtrees a few weeks ago and we love it!

Thank you!
